swift_cells.py

```diff
@@ -156,6 +157,8 @@ def __init__(
         # Open the input file
         with h5py.File(snap_filename % {"file_nr": 0}, "r") as infile:
 
+            self.snapshot_datasets = SnapshotDatasets(infile)
```

Comment:
There's some duplication here now because the I/O code already scans the snapshot files to see what datasets are present. I think it might make sense to remove the old code and use the SnapshotDatasets class instead.

The old code can handle missing meta data (e.g. if there are no stars or black holes in this snapshot but they do exist later) by specifying a reference snapshot to read meta data from. This might not be needed in future since swift now creates datasets even if they would have zero elements.